quickstack/manifests/backups.pp

```diff
@@ -0,0 +1,104 @@
+class backups (
+        $user		= $quickstack::params::backups_user,
+	$script_src	= $quickstack::params::backups_script_src,
+        $script_local   = $quickstack::params::backups_script_local_name,
+	$backups_dir    = $quickstack::params::backups_directory,
+        $log_file       = $quickstack::params::backups_log,
+        $ssh_key        = $quickstack::params::backups_ssh_key,
+        $sudoers_d      = $quickstack::params::backups_sudoers_d,
+        $cron_email     = $quickstack::params::backups_email,
+        $cron_hour	= $quickstack::params::backups_hour,
+        $cron_min 	= $quickstack::params::backups_min,
+) {
+
+    $script_dest = "${backups_dir}/scripts/${script_local}"
+
+    package { 'rsync':
+      ensure => installed,
+    }
+
+    user { "${user}":
+      ensure     => present,
+      comment    => 'backups user',
+      home       => "/home/${user}",
+      managehome => true,
+      before     => File[ 'ssh_dir', 'backup_script', 'sudo-permissions' ],
+     }
+
+    file { 'ssh_dir' :
+      ensure => directory,
+      path   => "/home/${user}/.ssh",
+      owner  => $user,
+      group  => $user,
+      mode   => '0700',
+      before => File['authorized_keys'],
+    }
+
+    file { 'authorized_keys':
+      ensure  => present,
+      path    => "/home/${user}/.ssh/authorized_keys",
+      owner   => $user,
+      group   => $user,
+      mode    => '0600',
+      replace => true,
+      source  => $ssh_key,
+      require => Class['ssh::server::install'],
+    }
+    
+   file { [ $backups_dir, "${backups_dir}/scripts" ] :
+      ensure => 'directory',
+      owner  => 'root',
+      group  => 'root',
+      before => File['backup_script'],
+    }
+
+    file { 'backup_script':
+      path    => $script_dest,
+      ensure  => file,
+      source  => $script_src,
+      owner   => 'root',
```

Comment:
$user's sudo is restricted (and one of the things on my to-do list is to lock that down even further), but the script is run by root.  If $user were compromised, I don't want an attacker editing the script to run their own commands as root.